lib/core/request.js

```diff
@@ -164,10 +164,18 @@ class Request {
         processHeader(this, headers[i], headers[i + 1])
       }
     } else if (headers && typeof headers === 'object') {
-      const keys = Object.keys(headers)
-      for (let i = 0; i < keys.length; i++) {
-        const key = keys[i]
-        processHeader(this, key, headers[key])
+      if (headers[Symbol.iterator]) {
+        for (const header of headers) {
```

Comment:
there should be an error if header isn't an array or its length isn't 2

Reply:
Is there necessity to explicitly mention array of two elements or could I just leave something like `Headers must be in key-value pair format.`?

Reply:
your suggestion is good

Reply:
Fixed